File: .net & mono/Snacks/OObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Noear.Snacks
{
    internal class OObject
    {
        public Dictionary<String, ONode> members;

        public OObject()
        {
            members = new Dictionary<String, ONode>();
        }

        public void set(String key, ONode value)
        {
            members[key] = value;
        }

        public ONode get(String key)
        {
            return members[key];
        }

        public bool contains(String key)
        {
            return members.ContainsKey(key);
        }

        public void clear() { }
    }
}

File: ios/Snacks/Snacks/src/OObject.swift
//
//  OObject.swift
//  snacks
//
//  Created by noear on 16/2/21.
//  Copyright © 2016年 noear. All rights reserved.
//

import Foundation

public class OObject {
    public final var members:[String:ONode] = [String:ONode]();
    
    public func set(_ key:String,_ value:ONode){
        members[key] = value;
    }
    
    public func get(_ key:String)->ONode{
        return members[key]!;
    }
    
    public func contains(_ key:String)->Bool{
        return members.contains(where: { (k,v) -> Bool in
            return key == k;
        });
    }
    
    public func count()->Int{
        return members.count;
    }
    
    public func clear(){
        return members.removeAll();
    }
}
